tidy3d/plugins/expressions/metrics.py

```diff
@@ -66,9 +66,9 @@ def evaluate(self, *args: Any, **kwargs: Any) -> NumberType:
         amps = (
             data[self.monitor_name]
             .amps.sel(direction=self.direction, mode_index=self.mode_index)
-            .isel(f=0)
+            .sel(f=self.freqs[0], method="nearest")
```

Comment:
I still dont quite understand the use of `0` here, doesn't this just ignore the frequency defined in the class?

Reply:
sure yeah we can use `self.freqs`. it's guaranteed to be a single frequency through the validator

Reply:
fixed

Reply:
if self.freqs is a single frequency only maybe it should just be `freq: float = None` (and if None we isel?)

Reply:
not sure i understand? multifrequency support will be added later. currently there is only a validator that checks that len(freqs)>1, which we can remove once multifrequency works

Reply:
or do you mean making freqs optional and then isel if only a single freq is found in the monitor? but that'll be a pain to validate

Reply:
this would make it consistent with the xarray coords too. I think this is basically something I missed when reviewing the original metrics PR

Reply:
because we do want to supert multiple freqs in the feature. it‘s just disabled right now

Reply:
but in xarray we also would just overload the singular coordinate with a list, eg `f=[f1, f2]`. So maybe we should follow this approach here?

Reply:
basically might be nice to be able to essentially copy the kwargs passed to `sel(**kwargs)` into `ModePower(**kwargs)`?

Reply:
ah, i guess i was trying to resemble the modemonitor interface more than the dataarray one with the metrics

Reply:
yea I think it should probably mirror the ModeAmpsDataArray, because the ModeMontior doesn't do `mode_index` for example? and this is essentially a wrapper for picking data out of the ModeAMpsDataARray right?

Reply:
yeah that's a good point. I think `**kwargs` will not be possible because we do need to validate all the fields, the `sel`s in the metrics won't be arbitrary. but we can definitely change the naming. might be better to do it in a separate PR though?